operators/src/processing/rasterization/mod.rs

```diff
@@ -0,0 +1,1082 @@
+                                                if distance <= radius {
+                                                    let [y, x] = grid_idx.0;
+                                                    Some((
+                                                        x as usize + y as usize * tile_size_x,
+                                                        gaussian(distance, stddev),
+                                                    ))
+                                                } else {
+                                                    None
+                                                }
+                                            })
+                                            .collect::<Vec<_>>(),
+                                    )
+                                })
+                                .for_each_with(send, |send, val| {
+                                    debug_assert!(send.send(val).is_ok());
+                                });
+                        });
+                    });
+
+                    while let Some(result) = receive.recv().await {
+                        for (index, value) in result {
+                            tile_data[index] += value;
+                        }
+                    }
+                }
+
+                Ok(RasterTile2D::new_with_tile_info(
+                    query.time_interval,
+                    tile_info,
+                    GridOrEmpty::Grid(
+                        Grid2D::new(tiling_strategy.tile_size_in_pixels, tile_data)
+                            .expect(
+                                "Data vector length should match the number of pixels in the tile",
+                            )
+                            .into(),
+                    ),
+                ))
+            });
+
+            Ok(tiles.boxed())
+        } else {
+            Ok(generate_zeroed_tiles(self.tiling_specification, query))
+        }
+    }
+}
+
+fn generate_zeroed_tiles<'a>(
+    tiling_specification: TilingSpecification,
+    query: RasterQueryRectangle,
+) -> BoxStream<'a, util::Result<RasterTile2D<f64>>> {
+    let tiling_strategy =
+        tiling_specification.strategy(query.spatial_resolution.x, -query.spatial_resolution.y);
+    let tile_shape = tiling_strategy.tile_size_in_pixels;
+
+    stream::iter(
+        tiling_strategy
+            .tile_information_iterator(query.spatial_bounds)
+            .map(move |tile_info| {
+                let tile_data = vec![0.; tile_shape.number_of_elements()];
+                let tile_grid = Grid2D::new(tile_shape, tile_data)
+                    .expect("Data vector length should match the number of pixels in the tile");
+
+                Ok(RasterTile2D::new_with_tile_info(
+                    query.time_interval,
+                    tile_info,
+                    GridOrEmpty::Grid(tile_grid.into()),
+                ))
+            }),
+    )
+    .boxed()
+}
```

Comment:
Wouldn't it be better to output EmptyTiles instead of outputting tiles with zeroes?

Reply:
It did before, but since the operator counts points, outputting a zero in case no points are present instead of "nothing" seemed more logical to me. Empty tiles are also not treated like zeroed tiles by the colorizer iirc. I think it is confusing to get a different result when a vector source does not contain any points (because it has a different type) versus when it does, but not in the queried region (or is actually empty, but still of the MultiPoint type).